diff -c old_src/libmecab.cpp src/libmecab.cpp
*** old_src/libmecab.cpp	Sun Apr 19 00:03:04 2009
--- src/libmecab.cpp	Wed Sep  9 23:46:37 2009
***************
*** 50,56 ****
  #if defined(_WIN32) && !defined(__CYGWIN__)
  HINSTANCE DllInstance = 0;
  
! #ifdef __cplusplus
  extern "C" {
  #endif
    BOOL WINAPI DllMain(HINSTANCE hinst, DWORD dwReason, LPVOID) {
--- 50,56 ----
  #if defined(_WIN32) && !defined(__CYGWIN__)
  HINSTANCE DllInstance = 0;
  
! #if defined(__cplusplus) && !defined(__MINGW32__)
  extern "C" {
  #endif
    BOOL WINAPI DllMain(HINSTANCE hinst, DWORD dwReason, LPVOID) {
***************
*** 62,68 ****
      std::locale::global(loc);
      return TRUE;
    }
! #ifdef __cplusplus
  }
  #endif
  #endif
--- 62,68 ----
      std::locale::global(loc);
      return TRUE;
    }
! #if defined(__cplusplus) && !defined(__MINGW32__)
  }
  #endif
  #endif
diff -c old_src/mecab.h src/mecab.h
*** old_src/mecab.h	Sun Apr 19 00:03:04 2009
--- src/mecab.h	Wed Sep  9 23:47:43 2009
***************
*** 128,134 ****
  extern "C" {
  #endif
  
! #ifdef _WIN32
  #include <windows.h>
  #  ifdef DLL_EXPORT
  #    define MECAB_DLL_EXTERN  __declspec(dllexport)
--- 128,134 ----
  extern "C" {
  #endif
  
! #if defined(_WIN32) && !defined(__MINGW32__)
  #include <windows.h>
  #  ifdef DLL_EXPORT
  #    define MECAB_DLL_EXTERN  __declspec(dllexport)
